.github/workflows/test_on_push.yml

```diff
@@ -52,3 +53,9 @@ jobs:
       - name: Test Template Generation
         run: |
           nox -s test-generation
+
+      - name: Upload coverage report
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version == '3.12'
+        uses: codecov/codecov-action@v4.5.0
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
Are we using the `coverage` module so far? I think we need to add it and integrate it with the tests first for any meaningful report.

Reply:
Since the workflow and the GitHub app have been added which brought the primary purpose, I think we should merge this first and see how it works for any of the other PRs, and then fix things as necessary.